src/gcode-machine-control.cc

```diff
@@ -534,6 +551,14 @@ const char *GCodeMachineControl::Impl::special_commands(char letter,
   case 119: mprint_endstop_status(); break;
   case 120: pause_enabled_ = true; break;
   case 121: pause_enabled_ = false; break;
+  case 181: {
+    const char *const after_pair = parser_->ParsePair(remaining, &letter, &value, msg_stream_);
+    if (remaining == NULL) set_lookahead(-1);
+    else if (letter == 'S') set_lookahead((unsigned)value);
```

Comment:
If you're casting the value here to unsigned already, you might be in for a surprise: if someone sets `M181 S-1`, then you're setting values to int-max here. But ideally we want set set look ahead function have a chance to complain that it is below or above the value. Now it is always above the value.